src/structure/viewer.ts

```diff
@@ -1565,4 +1615,163 @@ export class MoleculeViewer {
+        return {
+            min: this._viewer.addLabel(
+                JSON.stringify(min),
+                this._genColorBarValSpec(20, gradHeight + gradPosShift + 2)
+            ),
+            mid: this._viewer.addLabel(
+                JSON.stringify(mid),
+                this._genColorBarValSpec(20 + gradWidth / 2, gradHeight + gradPosShift + 2)
+            ),
+            max: this._viewer.addLabel(
+                JSON.stringify(max),
+                this._genColorBarValSpec(20 + gradWidth, gradHeight + gradPosShift + 2)
+            ),
+            prop: this._viewer.addLabel(
+                title,
+                this._genColorBarValSpec(20 + gradWidth / 2, gradHeight + gradPosShift + 16)
+            ),
+            grad: this._viewer.addLabel('.', {
+                position: new $3Dmol.Vector3(20, gradPosShift + 4, 0),
+                backgroundImage: gradImgPath,
+                borderColor: 'black',
+                borderOpacity: 1,
+                borderThickness: 1,
+                fontColor: 'white',
+                fontSize: 2,
+                fontOpacity: 0,
+                inFront: true,
+                useScreen: true,
+            }),
+        };
+    }
+
+    /** Generate a LabelSpec for the key values in the color bar */
+    private _genColorBarValSpec(xPosition: number, yPosition: number): $3Dmol.LabelSpec {
+        return {
+            alignment: 'topCenter',
+            position: new $3Dmol.Vector3(xPosition, yPosition, 0),
+            font: 'sans',
+            fontColor: 'black',
+            fontSize: 14,
+            showBackground: false,
+            inFront: true,
+            useScreen: true,
+        };
+    }
+
+    /** Create a gradient.png file for the color bar */
+    private _generateGradientImage(
+        palette: string,
+        width: number,
+        height: number
+    ): HTMLCanvasElement {
+        const canvas = document.createElement('canvas');
+        canvas.width = width;
+        canvas.height = height;
+        const mapColor = COLOR_MAPS[palette];
+
+        const ctx = canvas.getContext('2d');
+
+        if (!ctx) {
+            throw new Error('Could not get 2D context from canvas');
+        }
+
+        // Create a linear gradient from colorStart to colorEnd
+        const gradient = ctx.createLinearGradient(0, 0, width, 0);
+        for (let c = 0; c < mapColor.length; c++) {
+            gradient.addColorStop(Number(mapColor[c][0]), mapColor[c][1]);
+        }
+
+        ctx.fillStyle = gradient;
+        ctx.fillRect(0, 0, width, height);
+
+        return canvas;
+    }
+
+    /**
+     * This function is used to delete and/or add a color bar.
+     * It is a public function because it has to be called
+     * when another viewer is created or removed from the grid
+     * in order to update the size of the color bar.
+     */
+    public colorBarUpdate = (action: string, min?: number, max?: number) => {
```

Comment:
Because update is deleting and then adding new labels. So `updateColorbar()` would only be a little extension to `deleteColorbar()`. Therefore, I decided to rather make one single function.

Reply:
Indeed, we don't need to pass min/max values here. I fixed that.

Reply:
> Because update is deleting and then adding new labels. So updateColorbar() would only be a little extension to deleteColorbar(). Therefore, I decided to rather make one single function.

Nothing prevents you from writing

```ts
function updateColorbar() {
    this.deleteColorbar();
    // Do the rest
}
```

This way you keep a clean structure on what's happening, but also offer a nice API to users of the class.